compiler/x/objectfmt/FunctionCallData.hpp

```diff
@@ -0,0 +1,224 @@
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef TR_FUNCTIONCALLDATA_INCL
+#define TR_FUNCTIONCALLDATA_INCL
+
+#include "objectfmt/OMRFunctionCallData.hpp"
+#include "infra/Annotations.hpp"
+#include "runtime/Runtime.hpp"
+
+namespace TR { class Instruction; }
+namespace TR { class SymbolReference; }
+namespace TR { class Node; }
+namespace TR { class Register; }
+namespace TR { class RegisterDependencyConditions; }
+namespace TR { class CodeGenerator; }
+
+
+namespace TR
+{
+
+class OMR_EXTENSIBLE FunctionCallData : public OMR::FunctionCallDataConnector
+   {
+public:
+
+   /**
+    * APIs for non-helper function calls
+    */
+
+   /**
+    * @param[in] parm_methodSymRef : \c TR::SymbolReference of the function to call
+    * @param[in] parm_callNode : \c TR::Node this function call is associated with
+    * @param[in] parm_scratchReg : a scratch register to use to emit the dispatch sequence
+    * @param[in] parm_targetAddress : a specific function address to call that overrides the
+    *               address in the symbol reference; or 0 if none
+    * @param[in] parm_regDeps : \c TR::RegisterDependency structure for this call
+    * @param[in] parm_reloKind : the external relocation kind associated with this function call
+    * @param[in] parm_useSymInstruction : if true then use "Sym" form instructions that take a
+    *               \c TR::SymbolReference; if false then use equivalent form instructions without
+    *               the \c TR::SymbolReference parameter
+    * @param[in] parm_adjustsFramePointerBy : if non-zero, the adjustment to be made to the internal
+    *               virtual frame pointer by the instruction to perform the function call
+    * @param[in] parm_cg : \c TR::CodeGenerator object
+    * @param[in] parm_isCall : if true, emit a call instruction; if false, emit a jump instruction
+    */
+   FunctionCallData(
+         TR::SymbolReference *parm_methodSymRef,
+         TR::Node *parm_callNode,
+         TR::Register *parm_scratchReg,
+         uintptr_t parm_targetAddress,
+         TR::RegisterDependencyConditions *parm_regDeps,
+         int32_t parm_reloKind,
+         bool parm_useSymInstruction,
+         int32_t parm_adjustsFramePointerBy,
+         TR::CodeGenerator *parm_cg,
+         bool parm_isCall = true) :
```

Comment:
I guess this can be subjective, but to me the `TR::CodeGenerator*` is _always_ needed for these objects to be constructed properly in each overload of this constructor. Where as the other parameters may change. To me it makes most sense to put all the parameters which are consistently _always_ present first. Given these APIs are exclusively used in the codegen, we should just put the `TR::CodeGenerator*` parameter first.